sections/shared/modals/SelectTokenModal/SelectTokenModal.tsx

```diff
@@ -0,0 +1,242 @@
+		() =>
+			tokenBalances != null
+				? Object.values(
+						mapValues(tokenBalances, ({ balance, token }, symbol) => {
+							const { address } = token;
+
+							const price =
+								symbol === CRYPTO_CURRENCY_MAP.ETH
+									? get(coinGeckoPrices, [CoinGeckoPriceIds.ETH, 'usd'], null)
+									: get(coinGeckoTokenPrices, [address.toLowerCase(), 'usd'], null);
+
+							return {
+								currencyKey: symbol,
+								balance,
+								usdBalance: price != null ? balance.multipliedBy(price) : null,
+								token,
+							};
+						})
+				  )
+				: [],
+		[coinGeckoPrices, coinGeckoTokenPrices, tokenBalances]
+	);
+
+	const searchFilteredTokens = useDebouncedMemo(
+		() =>
+			assetSearch
+				? tokenBalancesWithPrices.filter(({ token: { name, symbol, address } }) => {
+						const assetSearchQueryLC = assetSearch.toLowerCase();
+
+						return (
+							name.toLowerCase().includes(assetSearchQueryLC) ||
+							symbol.toLowerCase().includes(assetSearchQueryLC) ||
+							address.toLowerCase() === assetSearchQueryLC
+						);
+				  })
+				: tokenBalancesWithPrices,
+		[tokenBalancesWithPrices, assetSearch],
+		DEFAULT_SEARCH_DEBOUNCE_MS
+	);
+
+	const tokensResults = useMemo(() => {
+		const tokens = orderBy(
+			assetSearch ? searchFilteredTokens : tokenBalancesWithPrices,
+			(token) =>
+				token.usdBalance != null ? token.usdBalance.toNumber() : token.balance.toNumber(),
+			'desc'
+		);
+
+		return tokensToOmit?.length
+			? tokens.filter(({ currencyKey }) => !tokensToOmit.includes(currencyKey))
+			: tokens;
+	}, [assetSearch, searchFilteredTokens, tokenBalancesWithPrices, tokensToOmit]);
+
+	const renderRow: ComponentType<ListChildComponentProps> = ({ index, style }) => {
+		const tokenResult = tokensResults[index];
+		const { currencyKey, token, balance, usdBalance } = tokenResult;
+
+		return (
+			<div key={currencyKey} style={style}>
+				<TokenRow
+					key={currencyKey}
+					onClick={() => {
+						onSelect(currencyKey);
+						onDismiss();
+					}}
+					totalValue={usdBalance ?? undefined}
+					{...{ balance, token, selectedPriceCurrency, selectPriceCurrencyRate }}
+				/>
+			</div>
+		);
+	};
+
+	return (
+		<StyledCenteredModal onDismiss={onDismiss} isOpen={true} title={t('modals.select-token.title')}>
+			<SearchContainer>
+				<AssetSearchInput
+					placeholder={t('modals.select-token.search.placeholder')}
+					onChange={(e) => {
+						setAssetSearch(e.target.value);
+					}}
+					value={assetSearch}
+					autoFocus={true}
+				/>
+			</SearchContainer>
+			<RowsHeader>
+				<span>
+					{assetSearch ? (
+						<span>{t('modals.select-token.header.search-results')}</span>
+					) : (
+						t('modals.select-token.header.all-tokens')
+					)}
+				</span>
+				<span>{t('modals.select-token.header.holdings')}</span>
+			</RowsHeader>
+			<StyledRowsContainer>
+				{isLoading ? (
+					<Loader />
+				) : tokensResults.length > 0 ? (
+					<AutoSizer>
+						{({ height, width }) => (
+							<FixedSizeList
```

Comment:
originally... I was showing the whole token list (over 900 tokens), so there was a good use case for using virtualization. Now only tokens with > 0 balance are shown.

This would still be useful if in the future the behaviour will change again for some reason... so I'd leave the virtualization here.